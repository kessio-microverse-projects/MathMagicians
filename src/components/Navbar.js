/* eslint-disable react/display-name  */
import React from 'react';
import {
  BrowserRouter as Router, Routes, Route, Link,
} from 'react-router-dom';
import Calculator from './Calculator';
import Quote from './Quote';
import Home from './Home';
import '../styles/navbar.css';

const Navbar = () => (
  <Router>
    <nav className="nav">
      <Link className="logo" to="/">Math Magicians</Link>
      <div className="navItems">
        <Link className="linkItem" to="/">Home</Link>
        <Link className="linkItem border" to="/quote">Quote</Link>
        <Link className="linkItem border" to="/calculator">Calculator</Link>
      </div>
    </nav>

    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/quote" element={<Quote />} />
      <Route path="/calculator" element={<Calculator />} />
    </Routes>
  </Router>

);
export default Navbar;
