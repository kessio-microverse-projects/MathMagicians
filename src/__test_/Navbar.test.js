import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

describe('snapshot test for Navbar Component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot('Navbar');
  });
});
