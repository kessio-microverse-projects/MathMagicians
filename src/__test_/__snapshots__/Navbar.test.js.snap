// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Home Component Matches the snapshot: Navbar 1`] = `
Array [
  <nav
    className="nav"
  >
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      Math Magicians
    </a>
    <div
      className="navItems"
    >
      <a
        className="linkItem"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="linkItem border"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
      <a
        className="linkItem border"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </div>
  </nav>,
  <div
    className="sectionContainer"
  >
    <div
      className="sectiontitle"
    >
      Welcome to Our Page
    </div>
    <div
      className="homeContent"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam mollis pharetra magna, tempor auctor est vulputate sed. Maecenas dolor dolor, mattis et erat at, tempus egestas eros. Etiam in facilisis lorem, ut maximus tellus. Praesent blandit massa sit amet nisl bibendum hendrerit. Maecenas in nibh ac tortor elementum tristique. Cras efficitur gravida vestibulum. Fusce pulvinar molestie tortor, in vehicula neque iaculis eget. Suspendisse enim turpis, fermentum in tincidunt sed, malesuada ut est. Cras varius euismod velit, et maximus metus. Donec tincidunt nec lectus eget elementum. Praesent fermentum fermentum nunc, eget blandit ante tincidunt vitae. Praesent nec lacus et orci ornare dignissim. Aenean sed sem dapibus, euismod massa vitae, dapibus nisi. Praesent pulvinar, felis a varius vulputate, eros elit aliquam quam, eu euismod metus mi quis est. Proin hendrerit metus malesuada, condimentum purus vitae, convallis nisl. Aenean mattis velit eget elit egestas, id cursus mi rutrum.
    </div>
    <div
      className="homeContent"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam mollis pharetra magna, tempor auctor est vulputate sed. Maecenas dolor dolor, mattis et erat at, tempus egestas eros. Etiam in facilisis lorem, ut maximus tellus. Praesent blandit massa sit amet nisl bibendum hendrerit. Maecenas in nibh ac tortor elementum tristique. Cras efficitur gravida vestibulum.
    </div>
  </div>,
]
`;

exports[`snapshot test for Navbar Component Matches the snapshot: Navbar 1`] = `
Array [
  <nav
    className="nav"
  >
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      Math Magicians
    </a>
    <div
      className="navItems"
    >
      <a
        className="linkItem"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="linkItem border"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
      <a
        className="linkItem border"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </div>
  </nav>,
  <div
    className="sectionContainer"
  >
    <div
      className="sectiontitle"
    >
      Welcome to Our Page
    </div>
    <div
      className="homeContent"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam mollis pharetra magna, tempor auctor est vulputate sed. Maecenas dolor dolor, mattis et erat at, tempus egestas eros. Etiam in facilisis lorem, ut maximus tellus. Praesent blandit massa sit amet nisl bibendum hendrerit. Maecenas in nibh ac tortor elementum tristique. Cras efficitur gravida vestibulum. Fusce pulvinar molestie tortor, in vehicula neque iaculis eget. Suspendisse enim turpis, fermentum in tincidunt sed, malesuada ut est. Cras varius euismod velit, et maximus metus. Donec tincidunt nec lectus eget elementum. Praesent fermentum fermentum nunc, eget blandit ante tincidunt vitae. Praesent nec lacus et orci ornare dignissim. Aenean sed sem dapibus, euismod massa vitae, dapibus nisi. Praesent pulvinar, felis a varius vulputate, eros elit aliquam quam, eu euismod metus mi quis est. Proin hendrerit metus malesuada, condimentum purus vitae, convallis nisl. Aenean mattis velit eget elit egestas, id cursus mi rutrum.
    </div>
    <div
      className="homeContent"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam mollis pharetra magna, tempor auctor est vulputate sed. Maecenas dolor dolor, mattis et erat at, tempus egestas eros. Etiam in facilisis lorem, ut maximus tellus. Praesent blandit massa sit amet nisl bibendum hendrerit. Maecenas in nibh ac tortor elementum tristique. Cras efficitur gravida vestibulum.
    </div>
  </div>,
]
`;
